File: TextClustering/src/com/yanjiuyanjiu/ml/kmeans/TextVector.java
package com.yanjiuyanjiu.ml.kmeans;

import com.yanjiuyanjiu.ml.vector.Vector;

/**
 * 文本向量.
 *
 * @author soulmachine@gmail.com
 */
public class TextVector {
	/** 文本文件的绝对路径. */
	protected final String file;
	/** 所属类别. */
	private final String catagory;
	/** 该文本转化成的向量. */
	protected Vector vector;

	/**
	 * 构造函数.
	 *
	 * @param file
	 *            文件路径
	 * @param catagory
	 *            本文件所属的类别
	 * @param vector
	 *            文本向量
	 */
	public TextVector(final String file, final String catagory,
			final Vector vector) {
		this.file = file;
		this.catagory = catagory;
		this.vector = vector;
	}

	public String getFile() {
		return file;
	}

	public String getCatagory() {
		return catagory;
	}

	public Vector getVector() {
		return vector;
	}
}

File: TextClusteringScala/src/main/scala/com/yanjiuyanjiu/ml/kmeans/TextVector.scala
/**
 * 文本向量.
 *
 * @author soulmachine@gmail.com
 * @date 2013-2-28
 * @version 0.1
 * @since scala 2.10
 */
package com.yanjiuyanjiu.ml.kmeans

import com.yanjiuyanjiu.ml.vector.Vector

/**
 * 文本向量.
 * @param file 文本文件的绝对路径
 * @param catagory 所属类别
 * @param vector 该文本转化成的向量
 * @author soulmachine@gmail.com
 */
class TextVector(val file: String, val catagory: String,
  val vector: Vector) {
}